### PR TITLE
Nazario -- HW1

### PR DESCRIPTION
**Pull request by nazarioa**

### CODE REVIEW

File: js/app.js [hunk 1]
```diff
@@ -49,10 +78,21 @@
...
 	// implement function here
+	const addToShoppingList = (item, list = []) => {
+		if (!list || !Array.isArray(list)) {
+			throw Error('Parameter list must be an Array');
+		}
+
+		if (!validateIsProduct(item)) {
+			throw Error(PRODUCT_ERROR_MESSAGE);
+		}
+		list.push(item);
```
Comment (nazarioa) [inline js/app.js:93]: `list.concat(item)` none destructive one liner

File: js/app.js [hunk 2]
```diff
@@ -86,6 +126,14 @@
 	*/
 
 	// implement function here
+	const removeFromShoppingList = (list = []) => {
+		if (!Array.isArray(list) || list.length === 0) {
+			return;
+		}
+
+		list.pop();
```
Comment (nazarioa) [inline js/app.js:134]: `list.slice(0,-1);`

File: js/app.js [hunk 3]
```diff
@@ -1,11 +1,27 @@
-(function() { // protect the lemmings!
+(function () { // protect the lemmings!
+	const PRODUCT_ERROR_MESSAGE = 'Object must be a valid product. Properties missing: item (string), price (number)';
```
Comment (sonylnagale) [inline js/app.js:2]: I love this const to provide an error message.

File: js/app.js [hunk 4]
```diff
@@ -1,11 +1,27 @@
...
+
+	/**
+	 * validateIsProduct - is used to validate that an object has a valid price and item property.
+	 * @param item
+	 * @returns {boolean}
+	 */
+	const validateIsProduct = function (item) {
```
Comment (sonylnagale) [inline js/app.js:9]: This function is really great. I appreciate thinking outside the box!

File: js/app.js [hunk 5]
```diff
@@ -23,6 +39,19 @@
 	*/
 
 	// implement function here
+	let newShoppingListItem = (item, price) => {
```
Comment (sonylnagale) [inline js/app.js:42]: Why would you use `let` here?

File: js/app.js [hunk 6]
```diff
@@ -86,6 +126,14 @@
 	*/
 
 	// implement function here
+	const removeFromShoppingList = (list = []) => {
+		if (!Array.isArray(list) || list.length === 0) {
+			return;
```
Comment (sonylnagale) [inline js/app.js:131]: Hm, but what are we returning here? Are we returning `list` or nothing?

File: js/app.js [hunk 7]
```diff
@@ -120,6 +168,14 @@
 	*/
 
 	// implement function here
+	const removeFirstItem = (list) => {
+		if (!Array.isArray(list) || list.length === 0) {
+			return;
+		}
+
+		return list.splice(1, list.length);
```
Comment (sonylnagale) [inline js/app.js:176]: I believe that we can work on this one a bit to make sure that `list.length` isn't overwhelming the splice.

File: js/app.js [hunk 8]
```diff
@@ -521,17 +650,32 @@
...
 				'price': 1.59
 			} (else throw error)
-			- return ONE list that contains items in 
+			- return ONE list that contains items in
 			items1 THEN items in items2 as a single array
 	*/
 
 	// implement function here
+	const combineLists = (items1 = [], items2 = []) => {
+		items1.forEach(item => {
```
Comment (sonylnagale) [inline js/app.js:665]: Is there a way we can be DRYer?

File: js/app.js [hunk 9]
```diff
@@ -692,17 +854,25 @@
 	*/
 
 	// implement function here
+	const computeSum = (list = []) => {
+		return list.reduce((accumulator, item) => {
```
Comment (sonylnagale) [inline js/app.js:858]: Have we defined accumulator?

**Comment by sonylnagale** [comment]
Great work! You definitely understand this material!

**Pull request closed without merging**